### ISSUE TITLE
waveform-compare fails on transcoded files

### ISSUE THREAD

**Issue by ssinyagin** [openpreserve/scape-xcorrsound#16]
hi,
I'm trying to utilize the xcorrsound for telephony QoS analysis, and I made a quick test: a WAV file is encoded into GSM codec, and then back into WAV. Both files sound identically to my ear, but waveform-compare reports a failure.
You can see the files in http://www.k-open.com/s/xcorrsound_001.zip
`OSR_us_000_0018_8k.wav` is the original file from http://www.voiptroubleshooter.com/open_speech/american.html
`x.wav` is the file after converting to GSM and back to WAV.

Or maybe I'm using the tool for the wrong purpose?

**Comment by jasn** [comment]
Hi.

I just ran the following:

[jasn@legato apps]$ ./waveform-compare
/home/jasn/Downloads/tmp/OSR_us_000_0018_8k.wav
/home/jasn/Downloads/tmp/x.wav --block-size=1 --threshold=0.94
--pad-short-block
Success
Offset: 0
Similarity: 0.942407
[jasn@legato apps]$

You may need to fiddle with the threshold value for your particular
application, 1 means totally identical and 0 means accept anything as
similar. For the particular instance you gave me 0.94 works, but try a few
cases out and turn the knob until it works. The --pad-short-block basically
just adds some silence at the end of your audio file if the clip length is
not divisible by the block length.

I see you also had some trouble compiling, this should now be fixed.
Basically you just needed to add c++11 flag for your compiler.

I hope this helps!

Regards,
Jesper

2015-02-26 11:19 GMT+01:00 Stanislav Sinyagin <notifications@github.com>:

**Comment by ssinyagin** [comment]
great, thanks a lot. I somehow overlooked the threshold option

**Comment by ssinyagin** [comment]
wow, cool, it also finds the silence offset :)
